src/status_im/extensions/ethereum.cljs

```diff
@@ -0,0 +1,39 @@
+(ns status-im.extensions.ethereum
+  (:require [status-im.utils.handlers :as handlers]
+            [re-frame.core :as re-frame]
+            [status-im.models.wallet :as models.wallet]
+            [status-im.utils.ethereum.abi-spec :as abi-spec]
+            [status-im.utils.fx :as fx]
+            [status-im.ui.screens.navigation :as navigation]))
+
+(handlers/register-handler-fx
+ :extensions/transaction-on-result
+ (fn [cofx [_ on-result id result method]]
+   (fx/merge cofx
+             (when on-result
+               {:dispatch (on-result {:error nil :result result})})
+             (navigation/navigate-to-clean :wallet-transaction-sent))))
```

Comment:
are you sure that the number of args provided to this function is correct? it expects 3 afaik, but receives only 2 in this case